Added support for returning extended subject headings with informatio…

EreMaijala opened the pull request:
…n regarding subject type and source and exposed it in the search API.

This PR is a proposal for adding subjectsExtended that would return more information about the subject strings. This was requested by a real-world API user and makes sense since the simple subjects may be too simple if you need to do something programmatically with the results, and this could be found useful also in the VuFind UI.

demiankatz commented on module/VuFind/src/VuFind/RecordDriver/SolrMarc.php:
```diff
@@ -105,7 +124,34 @@ public function getAllSubjectHeadings()
                     if (!empty($current)) {
-                        $retval[] = $current;
+                        if ($extended) {
+                            $sources = [
```
It probably makes sense to make this a class property as well, for easier customization/extension.

demiankatz commented on module/VuFind/src/VuFind/RecordDriver/Summon.php:
```diff
@@ -50,9 +50,15 @@ class Summon extends SolrDefault
      * @return array
      */
-    public function getAllSubjectHeadings()
+    public function getAllSubjectHeadings($extended = false)
```
I wonder if it would be cleaner to refactor all of this similar to the SolrMarc driver, so we have a property with Summon field names associated with types, and we iterate through it in a loop... that way we don't have to repeat the loop code four times with different variable names.

EreMaijala replied:
Yes, indeed. Now done and Summon tested.

EreMaijala commented:
I believe I've now addressed the review comments.

I thought about having default logic somewhere but I think that it's not worth it and it's better that you see in the actual driver code what it's supposed to return.

demiankatz commented:
Thanks again -- I think this is in good shape now. I added a few bits of test coverage before merging.
